### ISSUE TITLE
Determine all the things

### ISSUE THREAD

**Issue by lollar** [superposition-dev/discite#1]
Purpose:
To have a community-based application that provides users with a variety of different projects that they can complete either to gain more experience in development in general or if a user has development experience but wants to experiment with a new language. Solutions provided to the projects will be open to feedback from other users.

Here is my idea of a basic overflow...

A user can create a project that can be solved by other users.  The project instructions will be similar to a README and should be language agnostic. A project can be tagged with different skill levels (to be determined how that is decided). Projects can be upvoted or downvoted by other users.

Solutions will (at least initially) be a link to a git repository. Solutions can also be upvoted or downvoted by the community and constructive feedback can be provided to the solution submitter. We can look into tracking the commit hash at the point of each submission to give a better idea of refactoring that has been completed. 

@reecebisel post your ideas here and we can hash them out. :)

**Comment by lollar** [comment]
Backend will be written in elixir and will be JSON API

**Comment by reecebisel** [comment]
so I'm torn on whether or not we need to have a node/express side. The article you sent makes a lot of sense. But I don't love the idea of the front end just reaching straight to the api. Maybe its that I feel like that would expose the api endpoint. But maybe it won't I have no clue. I'm going to do more research in this area. ask some JS dudes I know.

**Comment by reecebisel** [comment]
I think I'm going to write the front end in react/redux. Doesn't seem like we need to use node for any backend stuff. Maybe express unless we go full single page app. Which I might look into, if I do that I plan on using React router.

**Comment by lollar** [comment]
Yeah I think that sounds like a good plan. I don't care whether we go with a SPA or not, whatever you think.
